### PR TITLE
Relax SemVer versions allowing backwards compatible updates

### PR DESCRIPTION
**Pull request by blaxill**
Currently blocks #472: I needed `simple_logger` in `oak_tests` to be updated and thought I'd do them all as SemVer spec requires that any upstream updates must be non-breaking to match. The `^0.x` matches could possibly be relaxed further.  

### Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [x] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: examples/abitest/abitest_common/Cargo.toml [hunk 1]
```diff
@@ -5,6 +5,6 @@ authors = ["David Drysdale <drysdale@google.com>"]
 edition = "2018"
 
 [dependencies]
-serde = { version = "1.0", features = ["derive"] }
+serde = { version = "^1.0", features = ["derive"] }
```
Comment (daviddrysdale) [inline examples/abitest/abitest_common/Cargo.toml]: Could/should this be `"^1" ?

Comment (blaxill) [reply examples/abitest/abitest_common/Cargo.toml]: Done

**Comment by tiziano88** [comment]
- I personally don't believe in semver at all, I'd rather specify all deps as "*", occasionally bump dependencies via `cargo update` and fix what breaks, if any. Would that work here?
- https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio says:
  > The string "0.1.12" is a semver version requirement. Since this string does not have any operators in it, it is interpreted the same way as if we had specified "^0.1.12", which is called a caret requirement.
  so I am not sure I understand why this change should have any observable effect?

**Comment by blaxill** [comment]
> * I personally don't believe in semver at all, I'd rather specify all deps as "*", occasionally bump dependencies via `cargo update` and fix what breaks, if any. Would that work here?
> * https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio says:
>   > The string "0.1.12" is a semver version requirement. Since this string does not have any operators in it, it is interpreted the same way as if we had specified "^0.1.12", which is called a caret requirement.
>   > so I am not sure I understand why this change should have any observable effect?

1) This is possible, as we check in `Cargo.lock` files, SemVer would be more conservative (still requires `cargo update` but if you believe in SemVer then nothing should break). I haven't checked to see what (if anything) breaks if moving everything to "*" 

2) Regarding this last point David had the same question- I believe the doc is misleading here, `"0.1.12"` is equivalent to `"^0.1.12"` but because of this particular instance, i.e. only for `"0.x.y"` == 0 prefix, 3 digits, caret specified. It definitely doesn't imply caret for all numbers such as "1" != "^1" which I've checked.

**Comment by tiziano88** [comment]
If "*" unblocks you, I'd suggest we just use that everywhere (which we are already doing, for the most part). Using anything else risks us being stuck on a old / deprecated version of any dependency, and not realising until we manually inspect all of them. Also semver by itself does not actually _guarantee_ API compatibility, in fact I have very rarely see it being observed strictly.

**Comment by blaxill** [comment]
> If "*" unblocks you, I'd suggest we just use that everywhere (which we are already doing, for the most part). Using anything else risks us being stuck on a old / deprecated version of any dependency, and not realising until we manually inspect all of them. Also semver by itself does not actually _guarantee_ API compatibility, in fact I have very rarely see it being observed strictly.

Done, pending any CI issues

**Pull request merged**